chia/full_node/coin_store.py

```diff
@@ -445,10 +446,15 @@ async def get_coin_states_by_ids(
                     coin_ids_db = tuple(batch.entries)
                 else:
                     coin_ids_db = tuple([pid.hex() for pid in batch.entries])
+
+                max_height_sql = ""
+                if max_height != uint32.MAXIMUM_EXCLUSIVE - 1:
+                    max_height_sql = f"AND confirmed_index<={max_height} AND spent_index<={max_height}"
```

Comment:
I understand that this might be the most practical solution to this. I just want to note that "burning in" arguments into the string like this will defeat python's query cache, requiring it to be recompiled every time (as long as the arguments differ). Perhaps not a big deal, especially not if this is only used in this one query you're adding.

The other obvious risk with doing things like this is that type-checking is weak in python, and if `max_height` somehow, somewhere, can be a string, you have sql injection attacks.

Reply:
> I understand that this might be the most practical solution to this. I just want to note that "burning in" arguments into the string like this will defeat python's query cache, requiring it to be recompiled every time (as long as the arguments differ). Perhaps not a big deal, especially not if this is only used in this one query you're adding.

Yeah probably not very relevant here since we anyway probably don't end up with the same queries, usually. 

> The other obvious risk with doing things like this is that type-checking is weak in python, and if `max_height` somehow, somewhere, can be a string, you have sql injection attacks.

Thanks, good thoughts to keep in mind! Generally i think the ways the `max_height` comes to us, via the peer API is safe due to the deserialisation of the messages and in case thats something which gets exposed to RPC we generally also convert to int and if not in some case its probably also not so critical since the RPC interface is not public.